Libplanet/Store/IStore.cs

```diff
@@ -85,6 +65,80 @@ void PutBlock<T>(Block<T> block, Address addressesMask)
             AddressStateMap states
         );
 
+        /// <summary>
+        /// Lookup a state reference, which is a <see cref="Block{T}.Hash"/>
+        /// that has the state of the <paramref name="address"/>.
+        /// </summary>
+        /// <param name="namespace">The namespace to lookup a state reference.
+        /// </param>
+        /// <param name="address">The <see cref="Address"/> to find lookup.
+        /// </param>
+        /// <param name="lookupFrom">The starting point to lookup a state
+        /// reference. Blocks after <paramref name="lookupFrom"/> are ignored.
+        /// </param>
+        /// <returns>A <see cref="Block{T}.Hash"/> which has the state of the
+        /// <paramref name="address"/>.</returns>
+        /// <typeparam name="T">An <see cref="IAction"/> class used with
+        /// <paramref name="lookupFrom"/>.</typeparam>
+        /// <seealso cref="StoreStateReference{T}"/>
+        HashDigest<SHA256>? LookupStateReference<T>(
+            string @namespace,
+            Address address,
+            Block<T> lookupFrom)
+            where T : IAction, new();
+
+        /// <summary>
+        /// Stores a state reference, which is a <see cref="Block{T}.Hash"/>
+        /// that has the state of the <see cref="Address"/> for each updated
+        /// <see cref="Address"/>es by the <see cref="Transaction{T}"/>s in the
+        /// <paramref name="block"/>.</summary>
+        /// <param name="namespace">The namespace to store a state reference.
+        /// </param>
+        /// <param name="addresses">The <see cref="Address"/>es to store state
+        /// reference.</param>
+        /// <param name="block">The <see cref="Block{T}"/> which has the state
+        /// of the <see cref="Address"/>.</param>
+        /// <typeparam name="T">An <see cref="IAction"/> class used with
+        /// <paramref name="block"/>.</typeparam>
+        /// <seealso cref="LookupStateReference{T}"/>
+        void StoreStateReference<T>(
+            string @namespace,
+            IImmutableSet<Address> addresses,
+            Block<T> block)
+            where T : IAction, new();
+
+        /// <summary>
+        /// Forks state references, which are <see cref="Block{T}.Hash"/>es that
+        /// have the state of the <see cref="Address"/>es, from
+        /// <paramref name="sourceNamespace"/> to
+        /// <paramref name="destNamespace"/>.
```

Comment:
To me `destinationNamespace` looks better.

Reply:
Changed.